src/types/next-auth.d.ts

```diff
@@ -10,6 +10,11 @@ declare module "next-auth" {
     firstName?: string | null;
     lastName?: string | null;
   }
+  declare module "next-auth/jwt" {
+    interface JWT {
+      profile?: User;
+    }
+  }
```

Comment:
This should not be nested under the other module. It should be on the outer level.

If you seperate the modules you might have an issue with User not being defined for the JWT module. In that case you can define the User type on the top level, (you will also have to redefine the import for User where it is used).